src/lib/annotations/annotations.html

```diff
@@ -273,20 +273,42 @@
       }
     },
 
+    // In attribute names, some characters (e.g. '$') are considered valid by
+    // the HTML parser even though they are considered invalid by the DOM,
+    // which references the XML spec's definition of a valid attribute name.
+    // When `_replaceSlotWithContent` finds an attribute named 'name$', it
+    // needs to replace it with an attribute named 'select$', however the DOM
+    // prevents this attribute from being created directly due to XML's
+    // narrower valid attribute name definition. To work around this, the HTML
+    // parser is used to create an element with an attribute named 'select$'
+    // and this single attribute node is cloned whenever
+    // `_replaceSlotWithContent` needs to create a new 'select$' attribute.
+    _select$Attr: (function () {
+      var div = document.createElement('div');
+      div.innerHTML = '<div select$>';
+      return div.childNodes[0].attributes.getNamedItem('select$');
+    })(),
+
     _replaceSlotWithContent: function(slot) {
       var content = slot.ownerDocument.createElement('content');
       while (slot.firstChild) {
         content.appendChild(slot.firstChild);
       }
       var attrs = slot.attributes;
       for (var i=0; i<attrs.length; i++) {
-        var attr = attrs[i];
-        content.setAttribute(attr.name, attr.value);
+        content.attributes.setNamedItem(attrs[i].cloneNode());
```

Comment:
Let's add a comment here explaining that cloning the attribute node like this allows attributes with invalid characters to be moved.

Reply:
Added.